### ISSUE TITLE
Running on a virtual machine!

### ISSUE THREAD

**Issue by Fazelesswhite** [isohuntto/openbay#120]
Hey guys, I have this great idea in mind! I need your help to do it, Some of you might've got the OpenBay up and running on VMware or something, well if you got that to work why not upload the vmware files as a torrent and let others also just open the virtual machine and run it just with a single click, I came up with this idea since I'm trying to get the openbay running for the past 3 day but haven't worked even though I tried multiple times! If anyone can do the idea that's mentioned above then a lot more people would also participate in this!!

**Comment by TPBT-OFFICIAL** [comment]
Ill probably have it in less then 24 hours.

**Comment by TPBT-OFFICIAL** [comment]
Ok so i got this almost working.
http://prntscr.com/5n58um
The website frontend is done now i just need to finish the database and the sphinx server :)

**Comment by Fazelesswhite** [comment]
Cool ;)

**Comment by Fazelesswhite** [comment]
Hey buddy, is it ready yet ?

**Comment by TPBT-OFFICIAL** [comment]
I took a bit more because while putting the database somehow it deleted itself dunno why but its fine now.
So what am i supposed to put in the torrent file? is it the image of the server?

**Comment by Fazelesswhite** [comment]
An image will be amazing!

**Comment by TPBT-OFFICIAL** [comment]
Done!
Link: https://tpbt.org/torrent/13134187/Openbay-Pirate-Bay-clone-VMWARE-IMAGE (PLEASE SEED)
Operating System: Ubuntu Server
Config file location: /var/www/protected/config/config.php
Sphinx config file: /etc/sphinxsearch/sphinx.conf
After starting the server do the follow commands:
service mysql restart
service apache2 restart
indexer --all
service sphinxsearch restart

LOGIN - SHH:
Username: root
Password: 123456p (change it please)

Login - Mysql:
Username: root
Password: 123456p (change it please)
(If you change the mysql password you need to change the configurations on Sphinx and the OpenBay script!)

That should be it, you can then access mysql on http://[IP ADDRESS/DOMAIN NAME]/phpmyadmin and the Openbay script at [IP ADDRESS/DOMAIN NAME]/

If theres any problems with this please contact me on admin@porfibra.eu or admin@miao.sx

**Comment by Fazelesswhite** [comment]
Man you're a life saver! I will seed till I fucking die :')

**Comment by TPBT-OFFICIAL** [comment]
I forgot to put the specs so ill put them now:
Ram: 2-3GB minimum really but you can try to run it on less if you want
I put on vmware 1cpu/2cores i don't really know how it works but it seemed to run a bit slow whenever i tried to run "indexer --all" it would take forever.

**Comment by Fazelesswhite** [comment]
I'll give it 4GB of ram and 2cpu/2cores, that should be more than enough :)

**Comment by TPBT-OFFICIAL** [comment]
Yes it should :)

**Comment by Fazelesswhite** [comment]
I'm getting an error when I try to run "indexer --all" 
Here is the error - ERROR: index 'opbtorrents': sql_connect: Can't connect to MySQL server on '127.0.0.1' (111) (DSN=mysql://root:***@127.0.0.1:3306/opb).

**Comment by TPBT-OFFICIAL** [comment]
did you do the command "service mysql restart" and ws it sucessful?

**Comment by Fazelesswhite** [comment]
I didn't change anything, I just wanted to see if it works.

**Comment by TheChiefCoC** [comment]
@Fazelesswhite you have to change it to make it work

**Comment by Fazelesswhite** [comment]
What do you mean?

**Comment by TPBT-OFFICIAL** [comment]
@Fazelesswhite can i see the complete log after you ran "indexer --all" ?

**Comment by Fazelesswhite** [comment]
Sure..

**Comment by TheChiefCoC** [comment]
share the sphinx.conf at /etc/sphinxsearch/

**Comment by TPBT-OFFICIAL** [comment]
Actually just tried myself and for some reason mysql stops itself.....

**Comment by Fazelesswhite** [comment]
What is the log files location? 

Here is the sphinx.conf : http://pastebin.com/5ucDAY6k

**Comment by Fazelesswhite** [comment]
So what'll we do?

**Comment by TPBT-OFFICIAL** [comment]
This is something that doesn't normaly happen to me so i don't know exactly how to fix it

**Comment by Fazelesswhite** [comment]
Oh, TheChiefCOC, have any idea what to do?

**Comment by TheChiefCoC** [comment]
@TPBT-OFFICIAL @Fazelesswhite i recommend changing sql_host to localhost instead of 127.0.0.1

**Comment by TPBT-OFFICIAL** [comment]
Actually just found out how to fix it i think! but i need you to test it out!

Ok so go to http://(ip)/phpmyadmin
then login and go to the opb database and delete everything thats inside
then go to the console of the server in root and type "mysql -p" and put the mysql passwd
after that type "use opb;" and then "source 1.sql" (1.sql is the database file in the home folder of root)
and after that you need to wait a few hours when it finishes restart mysql and then try indexer --all again

the reason is i didn't finish that when i was doing the vmware image and so the database is corrupted i guess ill be doing it myself if it works ill update the torrent file i promise!

**Comment by TheChiefCoC** [comment]
also can you upload sphinx.conf to ghostbin or something, pastebin is blocked in my country

**Comment by Fazelesswhite** [comment]
I did what you told and now I'm getting : 
ERROR: 
Failed to open file '1.sql', error: 2

**Comment by TPBT-OFFICIAL** [comment]
are you in the home directory?
type cd and then try those commands again

**Comment by Fazelesswhite** [comment]
Now the following shows up

mysql> source 1.sql
Query OK, 0 rows affected (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

ERROR 1046 (3D000): No database selected
Query OK, 0 rows affected (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

ERROR 1046 (3D000): No database selected
Query OK, 0 rows affected (0.00 sec)

ERROR 1046 (3D000): No database selected
ERROR 1046 (3D000): No database selected
ERROR 1046 (3D000): No database selected
ERROR 1046 (3D000): No database selected
ERROR 1046 (3D000): No database selected

**Comment by TPBT-OFFICIAL** [comment]
you need to type "use opb" first
then source 1.sql

**Comment by Fazelesswhite** [comment]
Oh shit I forgot! Now its running I guess, I'll message you when it's done...

**Comment by TPBT-OFFICIAL** [comment]
just click ctrl + c and it should stop and try again

**Comment by Fazelesswhite** [comment]
Yeah I did that and it started like 5 mins ago..how much time will it take?

**Comment by TheChiefCoC** [comment]
it takes a lot of time depending on the size of db . my 433 MB sql file took like 10 mins or more i guess, i left it and browsed the web lol

**Comment by TPBT-OFFICIAL** [comment]
the database is about 5.3GB and to me it took more then 8 hours so dunno....

**Comment by TheChiefCoC** [comment]
which database is it @TPBT-OFFICIAL

**Comment by Fazelesswhite** [comment]
Wait a second..the database is just 3.7 gb it shows..

**Comment by TPBT-OFFICIAL** [comment]
Yes i know but i think its kinda compressed when i uploaded it it shows 5.3gb

**Comment by Fazelesswhite** [comment]
Oh okay..I hope it gets done fast :)

**Comment by TheChiefCoC** [comment]
which database are you using @TPBT-OFFICIAL

**Comment by TPBT-OFFICIAL** [comment]
@TheChiefCoC I downloaded the database directly from Openbay remote server and made it avaiable here: https://tpbt.org/torrent/13134188/Openbay-FULL-DATABASE-SQL-FORMATE

**Comment by TheChiefCoC** [comment]
@TPBT-OFFICIAL are you also using that database on your site?

**Comment by Fazelesswhite** [comment]
I did everything, but still I'm getting 404 shit happens :(

**Comment by TPBT-OFFICIAL** [comment]
@TheChiefCoC Yes i am.
@Fazelesswhite im really sorry but ill try to fix everything i promise!

**Comment by Fazelesswhite** [comment]
How did you get everything up and running so smoothly? I'm so jealous of you man :')

**Comment by TPBT-OFFICIAL** [comment]
@Fazelesswhite I don't think it can be considered smooth since the /recent is extremely slow and i haven't found out why yet.....

**Comment by TPBT-OFFICIAL** [comment]
i also think i fixed it ill update the torrent soon!

**Comment by TPBT-OFFICIAL** [comment]
Yes its fixed:
root@ubuntu:~# indexer --all
Sphinx 2.1.9-id64-release (rel21-r4761)
Copyright (c) 2001-2014, Andrew Aksyonoff
Copyright (c) 2008-2014, Sphinx Technologies Inc (http://sphinxsearch.com)

using config file '/etc/sphinxsearch/sphinx.conf'...
indexing index 'opb_common'...
collected 7800319 docs, 531.8 MB
sorted 186.3 Mhits, 100.0% done
total 7800319 docs, 531797231 bytes
total 658.217 sec, 807934 bytes/sec, 11850.66 docs/sec
indexing index 'opbtorrents'...
collected 4561000 docs, 302.4 MB

**Comment by Fazelesswhite** [comment]
Even I get through the indexer --all.. but when I try to access it.. it fails..

**Comment by TPBT-OFFICIAL** [comment]
It doesn't want to connect to mysql for some reason....

**Comment by Fazelesswhite** [comment]
shit..any fix? :(

**Comment by Fazelesswhite** [comment]
Try using the remote DB? Does that work?

**Comment by TPBT-OFFICIAL** [comment]
the database works now but for some reason it deleted itself again for me.

**Comment by Fazelesswhite** [comment]
Oh..I wonder how other people got it to work..

**Comment by Fazelesswhite** [comment]
Please check it out when you're free..I really need it to be up and running..

**Comment by TheChiefCoC** [comment]
i recommend starting fresh with nginx :)

**Comment by Fazelesswhite** [comment]
^ Did you get it up and running? I tried with nginx like 5 times on Debian.. With a remote DB.. Can you help me set it up with local DB?

**Comment by TheChiefCoC** [comment]
ok so here is what I told you before @Fazelesswhite change the sql_host in Sphinx.conf from 127.0.0.1 to localhost and run the command
```
service sphinxsearch restart

```

**Comment by Fazelesswhite** [comment]
Can you please elaborate, I reinstalled a new copy of debian 7 wheezy!

**Comment by TheChiefCoC** [comment]
If you want , I can set it all up for you :) :+1: Just inbox me

**Comment by TPBT-OFFICIAL** [comment]
Well im hosting my website tpbt.org on 9 different servers:
www1 - softlayer - ubuntu
www2 - softlayer - ubuntu
www3 - softlayer - ubuntu
www4 - softlayer - ubuntu
www5 - softlayer - ubuntu
www6 - softlayer - ubuntu
www7 - OVH - ubuntu
www8 - Ecatel - ubuntu
Database/Sphinx - OVH - centos

**Comment by Fazelesswhite** [comment]
Wow!

**Comment by Fazelesswhite** [comment]
@TheChiefCoC You also use FreeBSD?

**Comment by TheChiefCoC** [comment]
I use Ubuntu 14.04 and Apache

**Comment by TheChiefCoC** [comment]
And I have 3 servers which I use to host

**Comment by TPBT-OFFICIAL** [comment]
Actually 10 servers i forgot (www8 ubuntu)

**Comment by Fazelesswhite** [comment]
I have a server which has 5.8 GB ram and a i5 processor hahah!

**Comment by Fazelesswhite** [comment]
I am trying to install FreeBSD now, but I don't know why the DHCP client gave me an error during installation and for the time being I gave a static IP but even after rebooting and editing the /etc/rc.conf to select DHCP, it just doesn't give me internet access! What will I do now?

**Comment by MihaiB** [comment]
I've made a Vagrantfile in #135 which starts up an Ubuntu VM and automatically configures it to run the openbay. It uses the remote MySQL server but installs & configures a local Sphinx because the remote Sphinx is currently down. Sphinx requires a lot of RAM so I've set the VM to 2.5GB.

You can use it to run the openbay in a VM locally but also to have a concrete example of a working, reproducible setup which you can do on your own servers. It's all done from a few short, easy to read scripts.
